### PR TITLE
Sonnet V1.2.8

### PR DESCRIPTION
**Pull request by ultrabear**
PR for Sonnet V1.2.8
We kinda need Url blacklisting sooner than later
1.2.9 will add the wanted db.list_enum call and stuff if pycord still isint out by then

### CODE REVIEW

File: libs/lib_loaders.py [hunk 1]
```diff
@@ -23,13 +25,13 @@
 from lib_db_obfuscator import db_hlapi
 from lib_sonnetconfig import CLIB_LOAD, GLOBAL_PREFIX, BLACKLIST_ACTION
 
-from typing import Dict, List, Union, Any, Tuple, Optional, Type, cast
+from typing import Any, Tuple, Optional, Union, cast, Type, Dict
```
Comment (Saragon4005) [inline libs/lib_loaders.py:28]: what the hell happened here?

File: libs/lib_loaders.py [hunk 2]
```diff
@@ -234,12 +236,12 @@ def generate_infractionid() -> str:
 def inc_statistics_better(guild: int, inctype: str, kernel_ramfs: lexdpyk.ram_filesystem) -> None:
 
     try:
-        statistics: Dict[str, int] = kernel_ramfs.read_f(f"{guild}/stats")
+        statistics: dict[str, int] = kernel_ramfs.read_f(f"{guild}/stats")
     except FileNotFoundError:
         statistics = kernel_ramfs.create_f(f"{guild}/stats", f_type=cast(Type[Dict[str, int]], dict))
```
Comment (Saragon4005) [inline libs/lib_loaders.py:241]: sort of confused here why `Dict` and not `dict`

File: libs/lib_parsers.py [hunk 3]
```diff
@@ -40,13 +42,27 @@ class user_parse_error(Exception):
...
 
 
+_urlFilter = re.compile(r"[^a-z0-9\-\.]+")
+
+
+def _compileurl(urllist: List[str]) -> str:
+
+    urllist = [_urlFilter.sub("", i).replace(".", r"\.") for i in urllist]
+
+    return "(https?://)(" + "|".join(urllist) + ")\W?"
```
Comment (Saragon4005) [inline libs/lib_parsers.py:52]: I'ma just trust you on this

**Review by Saragon4005 — CHANGES_REQUESTED** [review]

File: libs/lib_loaders.py (hunk 2 above)
Comment (ultrabear) [reply libs/lib_loaders.py:241]: 3.8 doesnt allow 3.9 syntax outside of `: type` and `-> type` annotations, and thats with `from __future__ import annotations`

File: libs/lib_loaders.py (hunk 1 above)
Comment (ultrabear) [reply libs/lib_loaders.py:28]: unimported 3.9 things, then reimported them when it broke

File: libs/lib_parsers.py (hunk 3 above)
Comment (ultrabear) [reply libs/lib_parsers.py:52]: regex is ez

File: cmds/cmd_automod.py [hunk 4]
```diff
@@ -198,14 +227,15 @@ async def list_blacklist(message: discord.Message, args: List[str], client: disc
 
     # Print blacklist
     formatted_pretty = "```json\n" + formatted.replace('\\\\', '\\') + "```"
-    if len(formatted_pretty) <= 2000:
+    if len(formatted_pretty) <= 2000 and not raw:
         await message.channel.send(formatted_pretty)
     else:
+        errmsg = ["Total Blacklist too large to be previewed", "--raw specified, file supplied"][raw]
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply cmds/cmd_automod.py:234]: added a comment (its fast oK)

**Review by Saragon4005 — APPROVED** [review]

**Comment by ultrabear** [comment]
# Changelog:
## Frontend:
- Change(set-viewinfractions): Improved grammar substantially
- Add(url-blacklist): Add url blacklist system
- Add(list-automod): --raw flag to dump to file
- Change(version-info): Discord.py renamed to Wrapper to support pycord going forwards

## Under the Hood:
- Change(cloader): Refactored to use memory safe string header implementation 
- Change(cloader): Now compiles with `-Wall -Wextra -Werror -O3`
- Add(LeXdPyK): Testing for ramfs functionality in CI
- Add(csv blacklists): Add validation of characters inputted to prevent user errors, add - rm flag to clear
- Add(csv blacklists): Overall cleanup on system
- Add(lib_compatibility): compat system to allow the bot to run on dpy/pycord 2.0
- Add(lib_loaders): datetime_unix call to make a datetime from a unix timestamp
- Add(lib_starboard): Starboard deduplication library to prevent non uniform functionality
- Add(LeXdPyK): LeXdPyK 1.4.4, faster and more flexible than 1.4
- Change(db_hlapi): Values that take a sentinel None now check for None, type checks now use isinstance

## Runtime Environment:
- If you used LeXdPyK 1.4.1 or 1.4.2 during the development of 1.2.8, you must swap to >=1.4.3 as there is a ramfs bug causing caching errors

## Bugs:
- Patch(RegEx blacklist): Multi capture group regex would error the automod parser, it no longer does

**Pull request merged**